cdap-cli/src/main/java/co/cask/cdap/cli/CLIMain.java

```diff
@@ -248,10 +265,63 @@ public static void main(String[] args) {
         cliMain.tryAutoconnect();
         cliMain.updateCLIPrompt(cliConfig.getClientConfig());
 
-        if (commandArgs.length == 0) {
+        boolean ranScript = false;
+        boolean brokeFromScript = false;
+        if (launchOptions.getScriptFile() != null) {
+
+          ranScript = true;
+
+          ConnectionConfig connectionConfig = clientConfig.getConnectionConfig();
+          URI baseURI = connectionConfig.getURI();
+          URI uri = baseURI.resolve("/" + connectionConfig.getNamespace());
+          String cliPrompt = "\ncdap (" + uri + ")> ";
+
+          BufferedReader br = null;
+          try {
+            String line;
+            br = new BufferedReader(new FileReader(launchOptions.getScriptFile()));
+            // br = Files.newReader(launchOptions.getScriptFile(), Charset.UTF_8);
+            while ((line = br.readLine()) != null) {
+              String ln = line.trim();
+              String[] lnArr = ln.split("#");
+              if (lnArr.length == 0) {
+                continue;
+              }
+              String cmd = lnArr[0].trim();
+              if ("".equals(cmd)) {
+                continue;
+              }
+              if (cmd.equals("break")) {
+                brokeFromScript = true;
+                break; // Break from script and drop into interactive mode
+              }
+              output.println(cliPrompt + cmd);
+              cli.execute(cmd, output);
+            }
+            output.println("\n");
+          } catch (IOException e) {
+            output.println("Couldn't read from script '" + 
+                           launchOptions.getScriptFile() + "'\n" + e.getMessage());
+          } finally {
+            try {
+              if (br != null) {
+                br.close();
+              }
+            } catch (IOException e) {
+              output.println("Couldn't close script '" + 
+                             launchOptions.getScriptFile() + "'\n" + e.getMessage());
+            }
+          }
+        }
+        if (commandArgs.length > 0 && !brokeFromScript) {
+          String commandLine = Joiner.on(" ").join(commandArgs);
+          String[] commandLns = commandLine.split(";");
```

Comment:
Also probably don't want to treat it as a command separator if it's inside quotes. E.g. send stream x "hello;world" or send stream x 'hello;world'